homepage.js

```diff
@@ -0,0 +1,160 @@
+const cartItems = document.getElementById('cart-items');
+const totalPrice = document.getElementById('total-price');
+const productList = document.getElementById('product-list');
+const closeSidebar = document.getElementById('close-sidebar');
+    let products = [];
+    let cart = [];
+    let currentPage = 1; 
+    const itemsPerPage = 5;
+    function fetchProducts() {
+      fetch('https://fakestoreapi.com/products')
+        .then((response) => response.json())
+        .then((data) => {
+          products = data;
+          displayProducts(products);
+          addPagination(products);
+        })
+        .catch((error) => {
+          console.error('Error fetching products:', error);
+        });
+    }
+    function displayProducts(products) {
+      productList.innerHTML = '';
+      const startIndex = (currentPage - 1) * itemsPerPage;
+      const endIndex = startIndex + itemsPerPage;
+      const productsToDisplay = products.slice(startIndex, endIndex);
+     productsToDisplay.forEach((product) => {
+     const productElement = document.createElement('product-element');
+     productElement.image = product.image;
+     productElement.title = product.title;
+     productElement.price = product.price;
+     productElement.setAttribute('data-id', product.id);
+     productList.appendChild(productElement);
+    });
+   }
+   function addPagination(products) {
+    const pageCount = Math.ceil(products.length / itemsPerPage);
+    const pagination = document.getElementById('pagination');
+    pagination.innerHTML = '';
+  
+    for (let i = 1; i <= pageCount; i++) {
+      const pageButton = document.createElement('button');
+      pageButton.className = 'page-button';
+      pageButton.textContent = i;
+      pageButton.addEventListener('click', () => {
+        currentPage = i;
+        displayProducts(products);
+      });
+      pagination.appendChild(pageButton);
+    }
+  }
+    fetchProducts();
+    productList.addEventListener('click', (event) => {
+      if (event.target.classList.contains('add-to-cart')) {

```

Comment:
This should be inside ProductElement class, all click event of this product should be inside the class itself